docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+
+The syntax for instantiating a `file` object is as follows:
+
+`file f = file("filename.csv", "<mode>");`
+
+In the example provided above, two arguments are fed into the `file(...)` argument: *filename* for reading, writing or both, and *mode*. The *mode* argument can be `r` for read only, `w` for write only, and `rw` for both.
+
+Example:
+
+`file f = file("test.csv", "rw");` will open the File named test.csv in the current directory for both reading and writing.
+  
+`file f = file("filename.csv", "<mode>", "<delim>");`
+
+An optional third argument *delim* may be provided to the constructor specifying a delimiter for reading. If the *delim* argument is not supplied it will default to `\n`.
+
+Example:
+
+`file f = file("Program.java", "rw", ";");` will open the File named `Program.Java` in the current directory for both reading and writing. Calls to `read()` will read in chunks of the file delimited by the `;` character.
+
+#### 5.1.4. `func`
+
+`func` objects reference functions and are treated as first class citizens. The structure of `func` variable declarations is as follows:
+
+`func <return type> <name> = (<parameters>) => { <function body> };`
+
+Where:
+ * `<return type>` is the type returned by the function
+ * `<name>` is the variable name of the function
+ * `<paramters>` are the expected parameters for the function
+ * `<function body>` is the body of the function
+ 
+Once a function has been assigined to a `func` type it becomes a "named function" that is callable using that name e.g `funcName();`
+
+A function that does not return anything has a return type of `void`.  The void return type allows for programmers to create functions that are useful for their side effects. 
+
+##### Paramaterization
+
+Named functions can be passed to other functions as a parameter as follows\:
+```
+func void saySomething = () =>{ print("something"); };
+func void doSomething = (func f) => { f(); };
+doSomething(saySomething);
+```
+
+##### Caveats
+
+* Fire does not support function overloading
+* Fire does not support genericity in functions
+
+#### 5.1.4. `array`
+
+The `array` type is a dynamic collection of elements. Inspired by AWK's associative arrays, an `array` collection maps keys of one type to values of one type. Keys and values do not have be the same type, but all keys must share the same type and all values must share the same type.
+
+The structure of `array` variable declarations is as follows:
+
+`array arr[<key_type>, <value_type>];`
+
+Example:
+
+`array arr[int, string];` 
+
+The assignment of variables has the following structure:
+
+`arr[<key_value>] = <element>;` 
+
+Example:
+
+ `arr[17] = "age17";`
+ 
+ Finally, a programmer can retrieve a value associated with a key with the below syntax:
+ 
+ `int element = arr[<key_value>];`
+ 
+ Example:
+ 
+ `int age = arr["age"];`
+ 
```

Comment:
you should make a note about whether or not you can support nested arrays. for example:

```array[int, array[int, string]]```
which would give you a 2d array

And if so, how would you do something like
```array[array[int, int], array[int, int]```

and is this allowed:
```array[void, void]```

or
```array[regx, file]```

or
```array[file, regx]```

Reply:
you should give more explicit type constraints about what may appear as a key and what may appear as a value. typically, the constraints are tighter for the keys, requiring that they are comparable